OpenRA.Game/Primitives/PlayerDictionary.cs

```diff
@@ -11,7 +11,7 @@ public struct PlayerDictionary<T> : IReadOnlyList<T>, IReadOnlyDictionary<Player
 		readonly T[] valueByPlayerIndex;
 		readonly Dictionary<Player, T> valueByPlayer;
 
-		public PlayerDictionary(World world, Func<Player, T> valueFactory)
+		public PlayerDictionary(World world, Func<Player, int, T> valueFactory)
```

Comment:
Could you provide a constructor that still has the old signature for convenience and just have it delegate to this one.